### PR TITLE
deployment: Add helm chart.

### PR DESCRIPTION
**Pull request by Creatone**
Signed-off-by: Paweł Szulik <pawel.szulik@intel.com>

### CODE REVIEW

**Comment by Creatone** [comment]
@bobbypage PTAL

File: deploy/helm/templates/daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          {{if .Values.perf}}- --perf_events_config=/etc/config/perf-config.json{{end}}
+          {{if .Values.disable_metrics}}- --disable_metrics={{ join "," .Values.disable_metrics}}{{end}}
+          {{if .Values.enable_metrics}}- --enable_metrics={{ join "," .Values.enable_metrics}}{{end}}
+          {{if .Values.port}}- --port={{.Values.port}}{{end}}
+          {{if .Values.docker_only}}- --docker_only={{.Values.docker_only}}{{end}}
+        securityContext:
+          {{- toYaml .Values.securityContext | nindent 10 }}
+        resources:
+          requests:
+            memory: 400Mi
```
Comment (bobbypage) [inline deploy/helm/templates/daemonset.yaml:39]: why did we pick these values?

File: deploy/helm/templates/service.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
```
Comment (bobbypage) [inline deploy/helm/templates/service.yaml:1]: is there a way to disable creating service? I would imagine not everyone wants to create a nodeport, so maybe should be configurable?

File: deploy/helm/values.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+enable_metrics:
+  - perf_event
```
Comment (bobbypage) [inline deploy/helm/values.yaml:2]: should we enable these by default? maybe they should be opt-in similar as normal cAdvisor?

File: deploy/helm/templates/daemonset.yaml (hunk 1 above)
Comment (Creatone) [reply deploy/helm/templates/daemonset.yaml:39]: Source:  https://github.com/google/cadvisor/blob/master/deploy/kubernetes/base/daemonset.yaml#L23

File: deploy/helm/values.yaml (hunk 3 above)
Comment (Creatone) [reply deploy/helm/values.yaml]: Yes, I agree, they should be opt-in similar to normal cAdvisor.

**Comment by Creatone** [comment]
@bobbypage @iwankgb @dims @mrunalp PTAL

**Comment by bobbypage** [comment]
I would prefer we avoid adding additional deployment modes into cAdvisor itself that we need to support long term (we already support basic kustomize (https://github.com/google/cadvisor/tree/master/deploy/kubernetes). Users can always create external repos for helm charts using the existing GCR images, and it looks like already a few exist: https://artifacthub.io/packages/helm/ckotzbauer/cadvisor

What do you think?

**Pull request closed without merging**